lib/tapioca/dsl/compilers/protobuf.rb

```diff
@@ -206,13 +206,13 @@ def create_descriptor_method(klass, desc)
 
           klass.create_method(
             field.name,
-            return_type: field.type
+            return_type: "T.nilable(#{field.type})"
           )
 
           klass.create_method(
             "#{field.name}=",
-            parameters: [create_param("value", type: field.type)],
-            return_type: field.type
+            parameters: [create_param("value", type: "T.nilable(#{field.type})")],
+            return_type: "T.nilable(#{field.type})"
```

Comment:
Assignment after instantiation works similar to return values, inconsistently with keyword args :(
 So you can _only_ assign `nil` to fields whose type is another Message type.

```ruby
[12] pry(main)> cart = Shopify::Protobuf::Cart.new
[13] pry(main)> cart.id = nil 
Google::Protobuf::TypeError: Invalid argument for string field 'id' (given NilClass).
from (pry):13:in 'method_missing'
[14] pry(main)> cart.shop_id = nil
Google::Protobuf::TypeError: Expected number type for integral field 'shop_id' (given NilClass).
from (pry):14:in 'method_missing'
19] pry(main)> cart.shipping_address = nil  # Message fields CAN be assigned nil
=> nil
[20] pry(main)> cart.shipping_address = Shopify::Protobuf::ShippingAddress.new
=> <Shopify::Protobuf::ShippingAddress: id: 0, shop_id: 0, verified: false>
[26] pry(main)> cart.shipping_address = nil  # And you can set it back to nil again
=> nil
```